keyboards/reversestudio/decadepad/decadepad.c

```diff
@@ -1,7 +1,20 @@
 #include "decadepad.h"
-bool led_update_kb(led_t led_state) {
-    if (led_update_user(led_state)){
-        writePin(D4, led_state.num_lock);
-    }
-    return true;
-}
\ No newline at end of file
+void matrix_init_kb(void) {
+  led_init_ports();
+  matrix_init_user(); 
+};
+
+void led_init_ports(void) {
+  setPinOutput(D4);
+}
+
+void led_set_kb(uint8_t usb_led) {
+  if (IS_LED_ON(usb_led, USB_LED_NUM_LOCK)) {
+    // Turn NUM Lock LED on
+    writePinLow(D4);
+  } else {
+    // Turn NUM Lock LED off
+    writePinHigh(D4);
+  }
+    led_set_user(usb_led);
+}
```

Comment:
This should really still use the new API
```suggestion
bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)){
        writePin(D4, led_state.num_lock);
    }
    return true;
}
```

Reply:
but in new API, it will not work,so I turn back it to Old one to try to fix it

Reply:
You sure its not just the lack of `setPinOutput`? Its basically called in the same way so I cant see how it would "not work".

Reply:
The logic may need to be inverted, as it's setting the pin low to turn the LED on, but otherwise, yeah. Looks fine to me.

Reply:
yep it just need to invrted to logic, but in my own test, simply invert is not function, maybe I use a wron way?

Reply:
or can somebody just give me a inverted code to try it

Reply:
I've edited zvecr's suggestion above so that it should work (basically just added `!` before the `led_state.num_lock` to invert it).

Reply:
thx bro, it works